read_buffer/src/lib.rs

```diff
@@ -199,6 +200,18 @@ impl Database {
             Some(partition) => {
                 let mut chunks = vec![];
                 for chunk_id in chunk_ids {
+                    let chunk = partition
+                        .chunks
+                        .get(chunk_id)
+                        .ok_or_else(|| Error::ChunkNotFound { id: *chunk_id })?;
+
+                    ensure!(
+                        chunk.has_table(table_name),
+                        TableNotFound {
+                            table_name: table_name.to_owned(),
```

Comment:
```suggestion
                            table_name,
```

I find the snafu way almost too magical sometimes;

The `TableNotFound` is some fancy type not the actual `Error::TableNotFound` that will automatically call `into()` on its fields -- so in this case the `table_name` field is not actually a `String` but something like `impl Into<String>`